School/delivery/http/school_delivery.go

```diff
@@ -0,0 +1,98 @@
+package http
+
+import (
+	"github.com/airbenders/profile/domain"
+	"github.com/airbenders/profile/utils/errors"
+	"github.com/airbenders/profile/utils/httputils"
+	"github.com/gin-gonic/gin"
+	"net/http"
+	"reflect"
+)
+
+// SchoolHandler struct
+type SchoolHandler struct {
+	u domain.SchoolUseCase
+}
+
+// NewSchoolHandler returns a new SchoolHandler
+func NewSchoolHandler(u domain.SchoolUseCase) *SchoolHandler {
+	return &SchoolHandler{u}
+}
+
+// SearchStudentSchool is an endpoint that returns schools that matches the name
+func (h *SchoolHandler) SearchStudentSchool(c *gin.Context) {
+	domainName, ok := c.GetQuery("domain")
+	if !ok || domainName == "" {
+		c.JSON(http.StatusBadRequest, errors.NewBadRequestError("please provide a query"))
+		return
+	}
+
+	ctx := c.Request.Context()
+	schools, err := h.u.SearchSchoolByDomain(ctx, domainName)
+	if err != nil {
+		switch v := err.(type) {
+		case *errors.RestError:
+			c.JSON(v.Code, v)
+			return
+		default:
+			c.JSON(http.StatusInternalServerError, errors.NewInternalServerError(err.Error()))
+			return
+		}
+	}
+
+	c.JSON(http.StatusOK, schools)
+}
+
+// SendConfirmationMail sends email to the client for school confirmation
+func (h *SchoolHandler) SendConfirmationMail(c *gin.Context) {
+	ctx := c.Request.Context()
+	email := c.Query("email")
+	if email  == "" {
+		c.JSON(http.StatusBadRequest, errors.NewBadRequestError("must provide a valid email"))
+		return
+	}
+	var school domain.School
+	err := c.ShouldBindJSON(&school)
+	if err != nil || reflect.DeepEqual(school, domain.School{}) {
+		c.JSON(http.StatusBadRequest, errors.NewBadRequestError("must provide valid data"))
+		return
+	}
+
+	err = h.u.SendConfirmation(ctx, &domain.Student{ID: "asd", FirstName: "Stella"}, email, &school)
```

Comment:
are confirmations only going to be sent to Stella?

Reply:
Haha, this was hardcoded because we didn't have auth service connected. This has been fixed in the next PR